modules/programs/hexchat.nix

```diff
@@ -0,0 +1,248 @@
+    xdg.configFile."hexchat" = mkIf (cfg.theme != null) {
+      source = cfg.theme.out;
+      recursive = true;
+    };
+    xdg.configFile."hexchat/hexchat.conf" = mkIf (cfg.settings != null) {
+      force = true;
+      text = let hexchatConf = cfg.settings;
+      in concatMapStringsSep "\n" (x: x + " = " + hexchatConf.${x})
+      (attrNames hexchatConf);
+    };
+    xdg.configFile."hexchat/servlist.conf" = mkIf (cfg.channels != null)
+      (if attrNames cfg.channels == [ ] then {
+        text = "";
+      } else {
+        force = true;
+        text = let
+          transformChannel = (channelName:
+            let
+              channel = cfg.channels.${channelName};
+              transformField = (k: v:
+                if v == "" then
+                  ""
+                else ''
+
+                  ${k}=${v}'');
+              listChar = c: l: concatMapStrings (transformField c) l;
+              loginMethodMap = {
+                nickServMsg = 1;
+                nickServ = 2;
+                challengeAuth = 4;
+                sasl = 6;
+                serverPassword = 7;
+                saslExternal = 10;
+                customCommands = 9;
+              };
+            in let
+              name = transformField "N" channelName;
+              nickname = transformField "I" channel.nickname;
+              nickname2 = transformField "i" channel.nickname2;
+              realName = transformField "R" channel.realName;
+              userName = transformField "U" channel.userName;
+              password = transformField "P" channel.password;
+              charset = transformField "E" channel.charset;
+              loginMethod = transformField "L"
+                (toString loginMethodMap.${channel.loginMethod});
+              servers = listChar "S" channel.servers;
+              autojoin = listChar "J" channel.autojoin;
+              commands = listChar "C" channel.commands;
+              options = let
+                computeFieldsValue = (with channel.options;
+                  (if channel.options == null then
+                    0
+                  else
+                    ((if autoconnect then 8 else 0)
+                      + (if !connectToSelectedServerOnly then 1 else 0)
+                      + (if !bypassProxy then 16 else 0)
+                      + (if useSSLForAllServers then 4 else 0)
+                      + (if acceptInvalidSSLCertificates then 32 else 0)
+                      + (if useGlobalUserInformation then 2 else 0))));
+              in transformField "F" (toString computeFieldsValue);
+              # Note: Missing option `D=`.
+            in name + loginMethod + charset + options + # `D=` position +
+            nickname + nickname2 + realName + userName + password + servers
+            + autojoin + commands);
+        in let
+          channels =
+            concatMapStringsSep "\n" transformChannel (attrNames cfg.channels);
+        in ''
+          v=2.14.3
```

Comment:
Good idea.

Reply:
Current config module is for that version and does not handle any other future version. Wouldn't it be a problem if it later upstream changes? By defining that version upstream will know the config format is an old one.

Reply:
I don't think limiting the version of this module is a good idea. It means that if program receive even a minor change, this will not work.

If this is the idea though, I would prefer an assertion, like:

```nix
{
  assertion = pkgs.hexchat.version == "2.14.3";
  message = "This module is compatible with only hexchat v2.14.3";
}
```

But I do think the above is a bad idea.